SPEC/FILES.md

```diff
@@ -759,6 +789,47 @@ A great source of [examples][] can be found in the tests for this API.
 
 The Mutable File System (MFS) is a virtual file system on top of IPFS that exposes a Unix like API over a virtual directory. It enables users to write and read from paths without having to worry about updating the graph. It enables things like [ipfs-blob-store](https://github.com/ipfs/ipfs-blob-store) to exist.
 
+#### `files.chmod`
+
+> Change mode for files and directories
+
+##### `ipfs.files.chmod(path, mode, [options])`
+
+Where:
+
+- `path` is the path to the entry to modify.  It might be:
+  - An existing MFS path to a file or a directory (e.g. `/my-dir/my-file.txt`)
+  - An IPFS path (e.g. `/ipfs/QmWGeRAEgtsHW3ec7U4qW2CyVy7eA2mFRVbk1nb24jFyks`)
+  - A [CID][cid] instance (e.g. `new CID('QmWGeRAEgtsHW3ec7U4qW2CyVy7eA2mFRVbk1nb24jFyks')`)
+- `mode` is the new file mode.  It might be:
+  - A string octal, e.g. `'0755'`
+  - A string modification of the existing mode, e.g. `'+x'`, `'-gw'`, etc
+  - An integer, e.g. the returned value from `parseInt('0755', 8)`
+- `options` is an optional Object that might contain the following keys:
+  - `recursive` is a Boolean value that indicates if `mode` should be applied to all sub files/directories of `path` (default: false)
+  - `hashAlg` is which algorithm to use when creating CIDs for modified entries. (default: `sha2-256`) [The list of all possible values]( https://github.com/multiformats/js-multihash/blob/master/src/constants.js#L5-L343)
+  - `flush` is a Boolean value to decide whether or not to immediately flush MFS changes to disk (default: true)
+  - `cidVersion`: the CID version to use for any updated entries (integer, default 0)
+
+**Returns**
+
+| Type | Description |
+| -------- | -------- |
+| `Promise<void>` | If action is successfully completed. Otherwise an error will be thrown |
+
+**Example:**
+
+```JavaScript
+// To give a file -rwxrwxrwx permissions
+await ipfs.files.chmod('/path/to/file.txt', parseInt('0777', 8))
+
+// Alternatively
+await ipfs.files.chmod('/path/to/file.txt', '+rwx')
```

Comment:
`?+rwx`?

Reply:
Not sure I follow - what does the leading `?` mean in `?+rwx`?  I can't see any reference to it in `man chmod` and when I try to do this locally I get an error:

```console
$ chmod ?+rwx foo.txt
chmod: Invalid file mode: ?+rwx
```

Reply:
Ah, nevermind. I didn't realize `+rwx` was equivalent to `a+rwx`.

Reply:
That's a good point, it will support `u+rwx`, `g+r`, `o+w`, `a-w` etc. though right? We should specify what subset of https://en.wikipedia.org/wiki/Modes_(Unix)#Format we do support.

Reply:
Yes, it'll support [all of those](https://github.com/ipfs/js-ipfs-mfs/pull/63/files#diff-e68b618aab9b596a8f2ed03586f2ef2aR79-R151).

Though now you mention it there's a testing gap for `a...`

..and maybe not `X` right now, though it can be added.